lib/Runtime/Base/FunctionInfo.cpp

```diff
@@ -64,7 +64,7 @@ namespace Js
     FunctionBody *
     FunctionInfo::GetFunctionBody() const
     {
-        return functionBodyImpl == nullptr ? nullptr : functionBodyImpl->GetFunctionBody();
+        return (functionBodyImpl && functionBodyImpl->IsFunctionBody()) ? functionBodyImpl->GetFunctionBody() : nullptr;
```

Comment:
Agreed.  I'd much rather you do that in JsDiagGetFunctionPosition and separately change the Assert in GetFunctionBody to fail fast instead.  Or change the API to always to TryGetFunctionBody which a check and make the API explicit, and eliminate all the extra check in the caller.